### PR TITLE
feat:🎁 add option to bootstrap filescoped namespace

### PR DESCRIPTION
**Pull request by cbosvik**
addresses #198

### CODE REVIEW

**Comment by GustavEikaas** [comment]
Would be nice if you can revert the styling/linting changes. The diff is pretty huge but its mostly lint changes

File: lua/easy-dotnet/options.lua [hunk 1]
```diff
@@ -137,14 +136,27 @@ local M = {
...
+  if type(a.auto_bootstrap_namespace) ~= "table" then
+    a.auto_bootstrap_namespace = {
+      type = "block_scoped",
+      enabled = a.auto_bootstrap_namespace == true
+    }
+  end
+end
+
 M.set_options = function(a)
   M.options = merge_tables(M.options, a or {})
```
Comment (GustavEikaas) [inline lua/easy-dotnet/options.lua:161]: missing the handle_auto_bootstrap_namespace here

Comment (cbosvik) [reply lua/easy-dotnet/options.lua:161]: Oh crap. Yeah I’ll fix that and those lint updates updates when I get back home.

Comment (GustavEikaas) [reply lua/easy-dotnet/options.lua:161]: No rush! Awesome work!♥

Comment (cbosvik) [reply lua/easy-dotnet/options.lua:163]: good morning! should be ok to merge now

**Review by GustavEikaas — DISMISSED** [review]

**Comment by GustavEikaas** [comment]
The readme had a lot of changes, maybe line endings idk. Anyways could you run the following command
`git restore . --source=feaa79ace82b10288ae3caf3f0c9e215088e3d0e` and just push the changes. It will make the diff nice and clean. There was also an outdated `important` in the readme that was removed in a previous PR

**Comment by cbosvik** [comment]
ok sorry. conform making a mess🙈

**Comment by GustavEikaas** [comment]
No worries, I havent set up my editor properly with lua formatting. Do you want to share your conform config for lua?

**Comment by cbosvik** [comment]
I found I spent too much time tweaking so I just use LazyVim nowadays :). Sometimes a little too much setup but generally sensible defaults. basically conform setup is done like so: 
return {
  "stevearc/conform.nvim",
  dependencies = { "mason.nvim" },
  lazy = true,
  cmd = "ConformInfo",
  keys = {
    {
      "<leader>cF",
      function()
        require("conform").format({ formatters = { "injected" }, timeout_ms = 3000 })
      end,
      mode = { "n", "v" },
      desc = "Format Injected Langs",
    },
  },
  init = function()
    -- Install the conform formatter on VeryLazy
    LazyVim.on_very_lazy(function()
      LazyVim.format.register({
        name = "conform.nvim",
        priority = 100,
        primary = true,
        format = function(buf)
          require("conform").format({ bufnr = buf })
        end,
        sources = function(buf)
          local ret = require("conform").list_formatters(buf)
          ---@param v conform.FormatterInfo
          return vim.tbl_map(function(v)
            return v.name
          end, ret)
        end,
      })
    end)
  end,
  opts = function()
    local plugin = require("lazy.core.config").plugins["conform.nvim"]
    local opts = {
      default_format_opts = {
        timeout_ms = 3000,
        async = false,
        quiet = false,
        lsp_format = "fallback",
      },
      formatters_by_ft = {
        lua = { "stylua" },
      },
      },
    }
    return opts
  end,
}

**Comment by cbosvik** [comment]
lol wifey does not approve I ruin the "uppesittarkväll" by staring at my computer. Need some family time. Ha en riktig go jul!

**Comment by GustavEikaas** [comment]
@cbosvik Family time is precious, en riktig god jul til deg og kona!

**Comment by cbosvik** [comment]
@GustavEikaas detsamme! Kjempeimponert av pluginet! Verktøyskassen ble øyblikkelig bedre, nesten komplett.

**Comment by GustavEikaas** [comment]
LGTM! Awesome. Lets merge?🚀

**Pull request merged**